### ISSUE TITLE
Navigaatio

### ISSUE THREAD

**Issue by thejhh** [norjs/mfw#1]
Aluksi tämä voi olla yksinkertainen menu -- joko vaaka tai pystysuunnassa (molemmat voi suunnitella, jolloin se voi olla automaattisesti dynaaminen), jossa auki oleva alisivu on korostettuna.

Toimintojen ikoneina voi käyttää esimerkiksi: https://fontawesome.com/v4.7.0/icons/ tai https://thenounproject.com/

**Comment by kaisis** [comment]
Onko tässä oikeaa ajatusta? Tämä ois siis se navigaatiopalkki. @thejhh 

![navigaatiopalkki vaaka](https://user-images.githubusercontent.com/33369580/80513803-46868500-8988-11ea-94f2-e2dc8b101419.png)

**Comment by thejhh** [comment]
Mitä toimintoja ajattelit näille ikoneille? 

Ajattelin tätä navigaatiota ikään kuin tabeiksi, joista yksi on valittuna ja mahdollisesti voisi olla tekstit mukana myös. Ainakin pystyversiossa. Tulisi näkyä helposti mikä on valittuna. 

Nuo viimeiset ikonit on ehkä liian leveitä, eivät oikein istu muiden kanssa samaan. Tosin voisivat toimia kun lyhentää tuon ylimääräisen leveyden. Molemmista sen voisi tehdä.

**Comment by kaisis** [comment]
Käsitin, että nämä tulis navigaatiopalkkiin: 
Sovelluksen asetukset
Ohjelmiston suunnittelutyökalu
Datamodel määrittely
Palveluiden määrittely
Näkymien määrittely
UI-komponentit
Vaatimusten määrittely
UI-komponentin muokkaus
Sovelluksen näkymän muokkaus
DataModelService
Arkkitehtuurin kuvaus
Button/Nappi
TEkstinsyöttö -UI-komponentit (input)

Olenko aivan hukassa :D

**Comment by thejhh** [comment]
Et ole, sinne juuri -- mutta tarkoitin vain, että missä järjestyksessä ja mille niistä :)

Tässä tiketissä kun sitä ei mainita.

Tästä listastahan siis navigaatiopalkkiin kuuluu vain nämä:

 * Sovelluksen asetukset
 * Data Model -määrittely
 * Palveluiden määrittely
 * Näkymien määrittely
 * Vaatimusten määrittely
 * UI-komponentit

Eli siis nuo feature-labelillä varustetut. Nämä eri päätoiminnot ovat siis "itsenäisiä" featureja, ja muut alinäkymät tulevat niiden alle. Poikkeuksena on tämä navigaationäkymä, joka on aina näkyvissä, ja sekin on oma featurensa.

Muut issuet ovat tämän projektin muita dokumentaatioita, tehtäviä, alinäkymiä tai teknisiä palveluita (näkymät siis käyttävät jaettuja palveluita juttujen varsinaiseen tekemiseen).

**Comment by thejhh** [comment]
Tuo index-label eli #3 taas kertoo koko sovellusprojektin koosteen. Siellä on listattu nuo navigaatioon tulevat osiot.

**Comment by kaisis** [comment]
Eli oisko tämä enemmän sinnepäin? 

![navigaatio auki v3](https://user-images.githubusercontent.com/33369580/80518574-4e95f300-898f-11ea-8379-25039aeaa3dc.png)

**Comment by thejhh** [comment]
Kyllä. Tuollaisesta esimerkkikuvasta on helppo lähteä toteuttamaan. :)

Sitten kun speksaa itsenäisempiä komponentteja voi sumentaa alueita, jotka ei ole tämän suunniteltavan asian kohteena. Navigaatiopalkissa ei toki ole isoja epäselvyyksiä siitä.

Tosin kannattaa tämmösessä UI-softassa mahdollisimman hyvin käyttää tila hyväksi. Tarkoitan siis, että tyhjää tilaa on toki syytä jättää paikkoihin mitä kosketaan tai klikataan (kuten nuo ikonit), siltä varalta jos sitä käytetään kosketusnäytöltä, mutta muulta osin suunnittelija haluaa koko näyttönsä hyötykäyttöön.

Välttämättä tekstejä tämä ei toki kaipaa. Suunnittelija oppii kyllä tunnistamaan mikä tarkoittaa mitä.
